### PR TITLE
Add two missing .Dispose invocations

### PR DESCRIPTION
**Pull request by joelverhagen**
@zhili1208 @emgarten

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
Issue # and impact?

**Comment by joelverhagen** [comment]
@yishaigalatzer no related issue number. I can create one if you want.

Impact prior to change: One is a leaked file nuspec file handle when writing lock file. The other is a non-disposed response when retrying after a HTTP 500+ is returned.

**Comment by joelverhagen** [comment]
> Why is the response dispose not in a finally statement?

The response is intended to leave the scope of this method, as this is the retry loop inside of `HttpSource` (which manages the eventual disposal). A dispose in a `finally` would always dispose it but the caller wants it in a non-disposed state.

This change only disposes a response that is not going to be used because it has an HTTP 500 and a new response is about to be fetched. 

If I am missing your point, please clarify.

File: src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRetryHandler.cs [hunk 1]
```diff
@@ -86,6 +86,8 @@ public HttpRetryHandler()
                             Strings.Http_RequestLog,
                             request.Method,
                             requestUri));
+                        
+                        response?.Dispose();
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRetryHandler.cs]: `Dispose` in catch block as well?
https://github.com/NuGet/NuGet.Client/pull/467/files#diff-c2ed78af63778dfdcfdbb72fd86300dbR116

Comment (alpaix) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRetryHandler.cs]: Talked to @joelverhagen offline. I think it's worth adding comments why it's disposed here and not say in finally.

Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpRetryHandler.cs]: Will do.

**Comment by emgarten** [comment]
Nice work tracking these down :shipit:

**Comment by alpaix** [comment]
:cool: :shipit:

**Pull request merged**